File: cs/XKS.Desktop/View/TableEditorView.cs
using Gtk;

namespace XKS.View
{
	public sealed class TableEditorView
	{
		public ListBox    TableList          { get; }
		public Toolbar    TableEditorToolbar { get; }
		public TreeView   TableEditor        { get; }
		public ToolButton AddColumnButton    { get; }

		public TableEditorView(ListBox    tableList,
		                       Toolbar    tableEditorToolbar,
		                       TreeView   tableEditor,
		                       ToolButton addColumnButton)
		{
			TableList = tableList;
			TableEditorToolbar = tableEditorToolbar;
			AddColumnButton = addColumnButton;
			TableEditor = tableEditor;
		}
	}
}

File: cpp/xks/view/TableEditorView.cpp
/**
 * Copyright 2020 Vitalijus Dobrovolskis
 *
 * Created by Vitalijus Dobrovolskis on 06.02.2020
 */

#include "TableEditorView.h"

xks::view::TableEditorView::TableEditorView(const RefPtr<Builder>& builder) : TableEditorView() {
	builder->get_widget("tableEditorToolbar", table_editor_toolbar);
	builder->get_widget("addColumnButton", add_column_button);
	builder->get_widget("tableEditor", table_editor_view);
	builder->get_widget("tableList", table_list);
}

Toolbar& xks::view::TableEditorView::get_table_editor_toolbar() const {
	return *table_editor_toolbar;
}

ToolButton& xks::view::TableEditorView::get_add_column_button() const {
	return *add_column_button;
}

TreeView& xks::view::TableEditorView::get_table_editor_view() const {
	return *table_editor_view;
}

ListBox& xks::view::TableEditorView::get_table_list() const {
	return *table_list;
}
